src/openrct2-ui/windows/Ride.cpp

```diff
@@ -4601,6 +4615,33 @@ static void window_ride_colour_invalidate(rct_window* w)
         window_ride_colour_widgets[WIDX_TRACK_ADDITIONAL_COLOUR].type = WindowWidgetType::Empty;
     }
 
+    // Selling item random colour checkbox
+    bool randomColourCheckboxIsVisible = false;
+    for (int item_index = 0; item_index < NUM_SHOP_ITEMS_PER_RIDE; item_index++)
+    {
+        const ShopItem *currItem = &rideEntry->shop_item[item_index];
```

Comment:
This looks a bit odd. You are creating a pointer, but on line 4625, you are referencing its value. It also doesn’t seem a pointer is necessary here.

I think you should be able to replace this with
```suggestion
        const ShopItem& currItem = rideEntry->shop_item[item_index];
```

and removing the asterisk on line 4625.